azurerm/internal/services/iothub/tests/iothub_enrichment_resource_test.go

```diff
@@ -0,0 +1,309 @@
+package tests
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMIotHubEnrichment_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_iothub_enrichment", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMIotHubEnrichmentDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMIotHubEnrichment_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMIotHubEnrichmentExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMIotHubEnrichment_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_iothub_enrichment", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMIotHubEnrichmentDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMIotHubEnrichment_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMIotHubEnrichmentExists(data.ResourceName),
+				),
+			},
+			{
+				Config:      testAccAzureRMIotHubEnrichment_requiresImport(data),
+				ExpectError: acceptance.RequiresImportError("azurerm_iothub_enrichment"),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMIotHubEnrichment_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_iothub_enrichment", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMIotHubEnrichmentDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMIotHubEnrichment_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMIotHubEnrichmentExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMIotHubEnrichment_update(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMIotHubEnrichmentExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
```

Comment:
Could we add a final basic step here?

Reply:
What  do you mean by a final basic step ? Is some checks missing ?

To be honest, it is my first PR on terraform (and first time with Go). I took inspiration from what has been done on the test from [iothub_route_resource_test.go](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/iothub/tests/iothub_route_resource_test.go)